lib/kadai-core-test/src/test/java/acceptance/taskrouting/TaskRoutingAccTest.java

```diff
@@ -74,41 +73,174 @@ void setUp(ClassificationService classificationService, WorkbasketService workba
+    class RouteTasksUsingRoutingTargetWithOwner implements KadaiConfigurationModifier {
+
+      @KadaiInject TaskService taskService;
+
+      @Override
+      public Builder modify(Builder builder) {
+        return builder.includeOwnerWhenRouting(true);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_When_TaskRouterReturnsEmptyRoutingTarget() {
+        Task task = taskService.newTask();
+        task.setClassificationKey(classificationSummary.getKey());
+        task.setPrimaryObjRef(DefaultTestEntities.defaultTestObjectReference().build());
+
+        assertThatThrownBy(() -> taskService.createTask(task))
+            .isInstanceOf(InvalidArgumentException.class)
+            .hasMessage("Cannot create a Task outside a Workbasket");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_SetWorkbasketAndOwnerForTask_When_TaskRouterDeterminesRoutingTarget()
+          throws Exception {
+        Task task = taskService.newTask(null, "DOMAIN_A");
+        task.setOwner("user-1-2");
+        task.setClassificationKey(classificationSummary.getKey());
+        task.setPrimaryObjRef(DefaultTestEntities.defaultTestObjectReference().build());
+
+        Task createdTask = taskService.createTask(task);
+
+        assertThat(createdTask.getWorkbasketSummary()).isEqualTo(domainAWorkbasket);
+        assertThat(createdTask.getOwner()).isEqualTo("user-1-1");
+
+      }
+    }
+
+    @Nested
+    @TestInstance(Lifecycle.PER_CLASS)
+    @WithServiceProvider(
+        serviceProviderInterface = TaskRoutingProvider.class,
+        serviceProviders = TaskRoutingProviderForRoutingTargetWithoutOwner.class)
+    class RouteTasksUsingRoutingTargetWithoutOwner implements KadaiConfigurationModifier {
+
+      @KadaiInject TaskService taskService;
+
+      @Override
+      public Builder modify(Builder builder) {
+        return builder.includeOwnerWhenRouting(true);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_When_TaskRouterReturnsEmptyRoutingTarget() {
+        Task task = taskService.newTask();
+        task.setClassificationKey(classificationSummary.getKey());
+        task.setPrimaryObjRef(DefaultTestEntities.defaultTestObjectReference().build());
+
+        assertThatThrownBy(() -> taskService.createTask(task))
+            .isInstanceOf(InvalidArgumentException.class)
+            .hasMessage("Cannot create a Task outside a Workbasket");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_SetWorkbasketAndKeepOwnerForTask_When_TaskRouterDeterminesRoutingTarget()
+          throws Exception {
+        Task task = taskService.newTask(null, "DOMAIN_A");
+        task.setOwner("user-1-2");
+        task.setClassificationKey(classificationSummary.getKey());
+        task.setPrimaryObjRef(DefaultTestEntities.defaultTestObjectReference().build());
+
+        Task createdTask = taskService.createTask(task);
```

Comment:
Just a question: how do I find out, which RoutingProvider is used?

Reply:
It's written above the class declaration next to "serviceProviders = "